### ISSUE TITLE
[BUG] Torch tensors using CUDA do not work with qml.ctrl

### ISSUE THREAD

**Issue by roboak** [PennyLaneAI/pennylane#4003]
### Expected behavior

qml.ctrl operation does not work on GPU. 

### Actual behavior

The code should run on GPU

### Source code

```shell
import pennylane as qml
import torch
dev = qml.device("default.qubit.torch", wires=2, torch_device='cuda')

@qml.qnode(dev, interface="torch")
def circuit(weight_1, weight_2):
    qml.ctrl(op=qml.RY(weight_1, wires=1), control=0) # when this gate is removed, all is fine
    qml.RX(weight_2, wires=0)
    return qml.state()

weights_1 = torch.tensor(0.1, device='cuda')
weights_2 = torch.tensor(0.1, device='cuda')

print(circuit(weights_1, weights_2))
# RuntimeError: Expected`

```
```


### Tracebacks

```shell
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0, CPU!
```

### System information

```shell
Name: PennyLane
Version: 0.28.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author:
Author-email:
License: Apache License 2.0
Location: c:\users\aksi01\miniconda3\envs\qns\lib\site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, retworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning, PennyLane-qiskit

Platform info:           Windows-10-10.0.19045-SP0
Python version:          3.8.15
Numpy version:           1.22.3
Scipy version:           1.7.3
Installed devices:
- qiskit.aer (PennyLane-qiskit-0.29.0)
- qiskit.basicaer (PennyLane-qiskit-0.29.0)
- qiskit.ibmq (PennyLane-qiskit-0.29.0)
- qiskit.ibmq.circuit_runner (PennyLane-qiskit-0.29.0)
- qiskit.ibmq.sampler (PennyLane-qiskit-0.29.0)
- lightning.qubit (PennyLane-Lightning-0.28.1)
- default.gaussian (PennyLane-0.28.0)
- default.mixed (PennyLane-0.28.0)
- default.qubit (PennyLane-0.28.0)
- default.qubit.autograd (PennyLane-0.28.0)
- default.qubit.jax (PennyLane-0.28.0)
- default.qubit.tf (PennyLane-0.28.0)
- default.qubit.torch (PennyLane-0.28.0)
- default.qutrit (PennyLane-0.28.0)
- null.qubit (PennyLane-0.28.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by timmysilv** [comment]
Thanks for reporting this! I believe #4002 will fix this once it is merged

**Comment by timmysilv** [comment]
the PR has been merged to master, and will be included in the upcoming 0.30 release
